test/clientconn_state_transition_test.go

```diff
@@ -616,3 +618,58 @@ func (s) TestConnectivityStateSubscriber(t *testing.T) {
 		}
 	}
 }
+
+// TestChannelStateWaitingForFirstResolverUpdate verifies the initial
+// state of the channel when a manual name resolver doesn't provide any updates.
+func (s) TestChannelStateWaitingForFirstResolverUpdate(t *testing.T) {
+	backend := stubserver.StartTestService(t, nil)
+	defer backend.Stop()
+
+	mr := manual.NewBuilderWithScheme("e2e-test")
+	defer mr.Close()
+
+	cc, err := grpc.NewClient(mr.Scheme()+":///", grpc.WithResolvers(mr), grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		t.Fatalf("Failed to create new client: %v", err)
+	}
+	defer cc.Close()
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	shortCtx, shortCancel := context.WithTimeout(ctx, defaultTestShortTimeout)
+	defer shortCancel()
+
+	cc.Connect()
+	testutils.AwaitState(shortCtx, t, cc, connectivity.Idle)
+
+	mr.UpdateState(resolver.State{
+		Addresses: []resolver.Address{{Addr: backend.Address}},
+	})
+
+	// Check that the channel transitions to CONNECTING then READY
+	testutils.AwaitState(shortCtx, t, cc, connectivity.Connecting)
+	t.Skipf("After connecting, before READY state: current state = %v\n", cc.GetState())
+
+	testutils.AwaitState(shortCtx, t, cc, connectivity.Ready)
+	t.Skipf("After entering READY state: current state = %v\n", cc.GetState())
+
+	mr.UpdateState(resolver.State{Addresses: nil})
+
+	done := make(chan struct{})
+	go func() {
+		_, err := testgrpc.NewTestServiceClient(cc).EmptyCall(shortCtx, &testgrpc.Empty{})
```

Comment:
This shouldn't use the `shortCtx` - we only use that for operations we expect to not happen.  We want this one to succeed.

Actually...literally everywhere you use `shortCtx` is wrong.  Use `ctx` instead.

But, we should use `shortCtx` in one check: before we produce the proper address from the resolver, and after we've asserted that we entered CONNECTING, we should do `testutils.AwaitNoStateChange(shortCtx, t, cc, connectivity.Connecting)`

Reply:
used ctx instead of shortCtx

Reply:
I don't see the AwaitNoStateChange.